shared/actions/chat/index.js

```diff
@@ -483,25 +474,8 @@ function _unboxedToMessage (message: ChatTypes.MessageUnboxed, yourName, yourDev
             throw new Error('empty attachment body')
           }
           const attachment: ChatTypes.MessageAttachment = payload.messageBody.attachment
-          const preview = attachment && attachment.preview
-          const mimeType = preview && preview.mimeType
-          const previewMetadata = preview && preview.metadata
-          const previewSize = previewMetadata && Constants.parseMetadataPreviewSize(previewMetadata)
-
-          const previewIsVideo = previewMetadata && previewMetadata.assetType === ChatTypes.LocalAssetMetadataType.video
-          let previewDurationMs = null
-          if (previewIsVideo) {
-            const previewVideoMetadata = previewMetadata && previewMetadata.assetType === ChatTypes.LocalAssetMetadataType.video && previewMetadata.video
-            previewDurationMs = previewVideoMetadata ? previewVideoMetadata.durationMs : null
-          }
-
-          const objectMetadata = attachment && attachment.object && attachment.object.metadata
-          const objectIsVideo = objectMetadata && objectMetadata.assetType === ChatTypes.LocalAssetMetadataType.video
-          let attachmentDurationMs = null
-          if (objectIsVideo) {
-            const objectVideoMetadata = objectMetadata && objectMetadata.assetType === ChatTypes.LocalAssetMetadataType.video && objectMetadata.video
-            attachmentDurationMs = objectVideoMetadata ? objectVideoMetadata.durationMs : null
-          }
+          const preview = attachment && (attachment.preview || attachment.previews && attachment.previews[0])
+          const attachmentInfo = Constants.getAttachmentInfo(preview, attachment && attachment.object)
```

Comment:
maybe this helper can mask these details about the preview vs previews. Maybe something like
`getAttachmentInfo(payload.messageBody)`